light-curve/light_curve/light_curve_py/features/rainbow/bolometric.py

```diff
@@ -0,0 +1,173 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Dict, List
+
+import numpy as np
+
+__all__ = ["bolometric_terms", "BaseBolometricTerm", "SigmoidBolometricTerm", "BazinBolometricTerm"]
+
+
+@dataclass()
+class BaseBolometricTerm:
+    """Bolometric term for the Rainbow"""
+
+    @staticmethod
+    @abstractmethod
+    def parameter_names() -> List[str]:
+        return NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def parameter_scalings() -> List[str]:
+        """Describes how to unscale the parameters - like time, timescale, flux or do not scale"""
+        return NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def value(self, t, params) -> float:
+        return NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def initial_guesses(t, m, sigma, band) -> Dict[str, float]:
+        return NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def limits(t, m, sigma, band) -> Dict[str, float]:
+        return NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def peak_time(params) -> float:
```

Comment:
What is the meaning of this value? Should it be infinite for a sigmoid?

Reply:
Well, it is supposed to be the true peak position - as for Bazin it is not the same as `reference_time` parameter. While for the end user it is often desirable to see where the peak actually is.
I agree it is not defined for the sigmoid - so for it I am keeping just the `reference_time` here, to have at least something indicative of the time when it changes (well, for sigmoid it is the mid-rise point, so does have at least some physical sense)

Reply:
Should we maybe change the name of the method? `reference_time`?

Reply:
Why? It is exactly the peak time (at least for Bazin, and for other bolometric terms that will have the peak) - while `reference_time` as used now in the parameters is not.

Reply:
Because of sigmoid

Reply:
But that's just a marginal degenerate case, where you obviously do not have a peak... If you wish I may just return `None` there, or raise `NotImplemented`... But better to have at least something reasonable here, in my opinion.

Reply:
Ok, let's keep it! I think it is a matter of documentation